include/openthread/thread.h

```diff
@@ -1009,6 +1009,14 @@ otError otThreadSendProactiveBackboneNotification(otInstance *              aIns
                                                   otIp6InterfaceIdentifier *aMlIid,
                                                   uint32_t                  aTimeSinceLastTransaction);
 
+/**
+ * This function detaches from the Thread network and forgets all credentials of the network.
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ *
+ */
+void otThreadLeaveNetwork(otInstance *aInstance);
```

Comment:
For routers, should send `ADDR_REL.ntf` to leader.
For sleepy end devices, I didn't find a official way to say goodbye to its parent in spec. I think it's useful to do so so that parent don't need to queue frames for the child, especially if the child has a very long timeout. How about we use `Child Update Request` to modify its Child Timeout to very short time so that its parent would immediately detect the change? @jwhui @abtink

Reply:
Yes, setting the Child Timeout value to a small value (even zero) is how a SED should say "goodbye".